### PR TITLE
GLK: SCOTT: Add support for ZX Spectrum games with graphics.

### PR DESCRIPTION
**Pull request by StableSteady**
Update the scott sub-engine to support ZX Spectrum games with graphics in .z80 file format. Also adds detection for the 11 Mysterious Adventures by Brian Howarth:

- The Golden Baton
- The Time Machine
- Arrow of Death Part 1 and 2
- Escape from Pulsar 7
- Circus
- Feasibility Experiment
- The Wizard of Akyrz
- Perseus and Andromeda
- Ten Little Indians
- Waxworks
Game source : http://www.ifarchive.org/indexes/if-archive/scott-adams/games/scottfree/

Questprobe: The Hulk is also fully supported but has no detection support because I couldn't source a legally available copy of the game.

### CODE REVIEW

File: engines/glk/scott/decompress_z80.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef GLK_SCOTT_DECOMPRESSZ80
+#define GLK_SCOTT_DECOMPRESSZ80
+
+#include <stdint.h>
```
Comment (dreammaster) [inline engines/glk/scott/decompress_z80.h:25]: There should be no inclusion of any standard library includes in ScummVM code. Anything the files need from them can probably be obtained from including common\scummsys.h, common\algorithm.h, or others.

**Comment by dreammaster** [comment]
Otherwise, looks good. Once it's resolved, I see no problem with merging it.

File: engines/glk/scott/decompress_z80.h (hunk 1 above)
Comment (StableSteady) [reply engines/glk/scott/decompress_z80.h]: Fixed. Using `common\scummsys.h` would have meant replacing every instance of `uint8_t` with `uint8` and so on for all types so I instead copied what the AGS engine does and created a new file `types.h` where I include `common\scummsys.h` and typedef over it.

Comment (dreammaster) [reply engines/glk/scott/decompress_z80.h]: One other point I just realized, for consistency, is that the commit prefixes should be "GLK: SCOTT: ", not "GLK SCOTT: ". Would you be able to do an interactive rebase?
If you do: git rebase -i HEAD~22
you should see a list of all the previous commits you did. If you then edit the "pick" before edit commit to be "r" or "reword", then when you save and quit it'll give you a chance to update each commit message in turn to add in the extra colon. Then do a push -f   only when you're happy with the result.

Comment (StableSteady) [reply engines/glk/scott/decompress_z80.h]: Done! 👍

**Pull request merged**